pkg/operator/controller/status.go

```diff
@@ -38,8 +40,14 @@ func (r *reconciler) syncOperatorStatus() error {
 		return fmt.Errorf("failed to get operator state: %v", err)
 	}
 
+	currentVersion := "unknown"
+	if co.Status.Versions != nil {
+		if len(co.Status.Versions[0].Version) != 0 {
+			currentVersion = co.Status.Versions[0].Version
```

Comment:
It would be better not to assume index position; instead, do a `for version := range co.Status.Versions` and look for one with `version.Name == "operator"`.